src/muya/lib/utils/exportHtml.js

```diff
@@ -222,12 +249,103 @@ class ExportHtml {
   <style>${extraCss}</style>
 </head>
 <body>
-  <article class="markdown-body">
   ${html}
-  </article>
 </body>
 </html>`
   }
+
+  /**
+   * @private
+   *
+   * @param {string} html The converted HTML text.
+   * @param {*} options The export options.
+   */
+  _prepareHtml (html, options) {
+    const { header, footer } = options
+    const appendHeaderFooter = !!header || !!footer
+    if (!appendHeaderFooter) {
+      return MARKDOWN_ARTICLE(html)
+    }
+
+    if (!options.extraCss) {
+      options.extraCss = footerHeaderCss
+    } else {
+      options.extraCss = footerHeaderCss + options.extraCss
+    }
+
+    let output = HF_TABLE_START
+    if (header) {
+      output += HF_TABLE_HEADER(options)
+    }
+
+    if (footer) {
+      output += HF_TABLE_FOOTER()
+      output = FIXED_FOOTER(options) + output
+    }
+
+    output = output + HF_TABLE_BODY(html) + HF_TABLE_END
+    return sanitize(output, EXPORT_DOMPURIFY_CONFIG)
+  }
+}
+
+const HF_TABLE_START = '<table class="page-container">'
+const HF_TABLE_BODY = html => {
+  return `<tbody><tr><td>
+  <div class="main-container">
+    ${MARKDOWN_ARTICLE(html)}
+  </div>
+</td></tr></tbody>`
+}
+const HF_TABLE_END = '</table>'
+
+const HF_TABLE_HEADER = options => {
+  const { header, headerFooterStyled } = options
+  const { type, left, center, right } = header
+  let headerClass = type === 1 ? 'single' : ''
+  headerClass += getHeaderFooterStyledClass(headerFooterStyled)
+  return `<thead class="page-header ${headerClass}"><tr><th>
+  <div class="hf-container">
+    <div class="header-content-left">${left}</div>
+    <div class="header-content">${center}</div>
+    <div class="header-content-right">${right}</div>
+  </div>
+</th></tr></thead>`
+}
+
+// Fake footer to reserve space.
+const HF_TABLE_FOOTER = () => {
+  return `<tfoot class="page-footer-fake"><tr><td>
+  <div class="hf-container">
+    &nbsp;
+  </div>
+</td></tr></tfoot>`
+}
+
+// The real footer at is shown at the bottom.
+const FIXED_FOOTER = options => {
+  const { footer, headerFooterStyled } = options
+  const { type, left, center, right } = footer
+  let footerClass = type === 1 ? 'single' : ''
+  footerClass += getHeaderFooterStyledClass(headerFooterStyled)
+  return `<div class="page-footer ${footerClass}">
+  <div class="hf-container">
+    <div class="footer-content-left">${left}</div>
+    <div class="footer-content">${center}</div>
+    <div class="footer-content-right">${right}</div>
+  </div>
+</div>`
+}
+
+const MARKDOWN_ARTICLE = html => {
```

Comment:
Why do you use Uppercase name of this function?

Reply:
It was const before :smile:

Reply:
It seems strange, we only use UPPERCASE name in vuex action now.